samples/dotnet/Startup.cs

```diff
@@ -0,0 +1,70 @@
+﻿using Microsoft.Owin;
+using Microsoft.Owin.Security;
+using Microsoft.Owin.Security.Cookies;
+using Microsoft.Owin.Security.OpenIdConnect;
+using Owin;
+using System;
+using System.Text;
+
+[assembly: OwinStartup(typeof(dotnet.Startup))]
+namespace dotnet
+{
+    public partial class Startup
+    {
+        public void Configuration(IAppBuilder app)
+        {
+            ConfigureAuth(app);
+            Configure(app);
+        }
+
+        private void Configure(IAppBuilder app)
+        {
+            app.Map("/error", HandleError);
+            app.Map("/logout", HandleLogout);
+
+            app.Run(async context =>
+            {
+                if (context.Authentication.User == null || context.Authentication.User.Identity == null || !context.Authentication.User.Identity.IsAuthenticated)
+                {
+                    // This is what [Authorize] calls   
+                    context.Authentication.Challenge(new AuthenticationProperties { RedirectUri = "/" }, OpenIdConnectAuthenticationDefaults.AuthenticationType);
+                    return;
+                }
+
+                // Return json structured user info for our tests.
+                context.Response.ContentType = "text/json";
+                StringBuilder rr = new StringBuilder("{");
+                foreach (var claim in context.Authentication.User.Claims)
+                {
+                    rr.Append(string.Format("\"{0}\":\"{1}\",", claim.Type, claim.Value));
+                }
+
+                rr.Remove(rr.Length - 1, 1);
```

Comment:
Isn't any better way to create JSON instead of concatenation of strings?

Reply:
In this case, no. This is a specific structure of the response for the tests.